assets/js/components/legacy-notifications/dashboard-modules-alerts.js

```diff
@@ -20,91 +20,86 @@
  * External dependencies
  */
 import { each } from 'lodash';
+import { useMount } from 'react-use';
 
 /**
  * WordPress dependencies
  */
-import { Component, Fragment } from '@wordpress/element';
+import { Fragment, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import { getModulesData } from '../../util';
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import NotificationAlertSVG from '../../../svg/notification-alert.svg';
 import Notification from './notification';
 import { modulesNotificationsToRequest, getModulesNotifications } from './util';
-import NotificationAlertSVG from '../../../svg/notification-alert.svg';
+const { useSelect } = Data;
 
-class DashboardModulesAlerts extends Component {
-	constructor( props ) {
-		super( props );
+function DashboardModulesAlerts() {
+	const [ alerts, setAlerts ] = useState( [] );
 
-		this.state = {
-			data: false,
-		};
-	}
+	const modules = useSelect( ( select ) =>
+		select( CORE_MODULES ).getModules()
+	);
 
-	componentDidMount() {
-		const modules = modulesNotificationsToRequest();
-
-		if ( modules ) {
-			getModulesNotifications( modules ).then( ( response ) => {
-				this.setState( {
-					data: response.results,
-				} );
-			} );
-		}
-	}
+	useMount( async () => {
+		const modulesWithNotifications = modulesNotificationsToRequest();
```

Comment:
There's been a regression (although not from this PR) where this function will return an empty array now. This is from the filter that was removed in https://github.com/google/site-kit-wp/commit/8d9b8102864c843accabc937190aa47cf21e90c6#diff-47b6b8d7a6c2934687704e4fba19f328306558821fe7836f023021b99cd26d69L61-L67 which should have been kept. AdSense is the only module that uses/implements notifications right now.